### PR TITLE
Update gsl develop from develop 2021/08/24

### PR DESCRIPTION
**Pull request by climbfuji**
Update gsl/develop from develop as of 2021/08/24.

Changes in this PR:

- Cmake build system updates
- Bugfix for wrong dimensions of variable `zorl` in surface restart files
- Remove nemsio
- Updates to coupled fields (ESMF)
- Update on how to pass MPI communicator to fv3atm (do not import from dycore)
- Update to inline post for regional runs
- Fix memory leak in stochastic physics
- Clean up redundant surface variables (does not change answer)
- Clean up grandfathered code, i.e. pass constants through argument list (changes answer only for csawmg tests)
- Bug fix for chemistry tracers (changes answers for gocart test only)
- Merge HAFS updates into main
- Thompson MP inner loop (alternative to subcycling, cannot be used together)
- Update of CCPP standard names and definition of Fortran derived data types in CCPP metadata
- Remove scheme `rayleigh_damp` from all suite definition files (wasn't used anyway)

Associated PRs:

https://github.com/NOAA-GSL/ccpp-physics/pull/104
https://github.com/NOAA-GSL/ccpp-framework/pull/17
https://github.com/NOAA-GSL/fv3atm/pull/106
https://github.com/NOAA-GSL/ufs-weather-model/pull/100

For regression testing, see https://github.com/NOAA-GSL/ufs-weather-model/pull/100

### CODE REVIEW

**Review by hannahcbarnes — APPROVED** [review]

File: stochastic_physics/stochastic_physics_wrapper.F90 [hunk 1]
```diff
@@ -294,9 +294,7 @@ subroutine stochastic_physics_wrapper (GFS_Control, GFS_Data, Atm_block, ierr)
          endif ! lndp block
```
Comment (tanyasmirnova) [inline stochastic_physics/stochastic_physics_wrapper.F90:294]: @ClaraDraper-NOAA @DomHeinzeller Dom, could we add to this PR a bug fix from Clara Draper needed for RRFS application.
Replace lines 250-254:
if (mod(GFS_Control%kdt,GFS_Control%nscyc) == 1 ) then
               param_update_flag = .true.
             else
               param_update_flag = .false.
             endif

with lines:
              if (GFS_Control%nscyc > 0) then 
                  if (mod(GFS_Control%kdt,GFS_Control%nscyc) == 1 ) then
                    param_update_flag = .true. !  update each time  gcycle has been called
                  else
                    param_update_flag = .false.
                  endif
              elseif ( (GFS_Control%nscyc == 0) .and. (GFS_Control%kdt==1) ) then
                 param_update_flag = .true. ! update at first time step if not calling gcyle (nscyc=0)
              else 
                 param_update_flag = .false.
                 endif

**Review by tanyasmirnova — COMMENTED** [review]
@DomHeinzeller @ClaraDraper-NOAA Dom and Clara, I think it is a good opportunity for a bug fix in stochastic wrapper that is needed for RRFS application. What do you think? Thank you!

File: stochastic_physics/stochastic_physics_wrapper.F90 (hunk 1 above)
Comment (ClaraDraper-NOAA) [reply stochastic_physics/stochastic_physics_wrapper.F90:294]: Works for me, if Dom is OK with it.

**Comment by DomHeinzeller** [comment]
This must be a separate PR, because it has nothing to do with updating gsl/develop from the authoritative repositories.

**Review by tanyasmirnova — APPROVED** [review]

**Review by DomHeinzeller — APPROVED** [review]
ccpp-physics hash 1263361 and ccpp-framework hash 5e410f3 are correct.

**Pull request merged**